crates/revm/src/optimism/rollup_cost.rs

```diff
@@ -0,0 +1,39 @@
+//! This contains a struct, [`RollupCostData`], that is used to compute the data availability costs
+//! for a transaction.
+
+use crate::optimism::fast_lz::flz_compress_len;
+
+/// RollupCostData contains three fields, which are used depending on the current optimism fork.
+///
+/// The `zeroes` and `ones` fields are used to compute the data availability costs for a
+/// transaction pre-fjord.
+///
+/// The `fastlz_size` field is used to compute the data availability costs for a transaction
+/// post-fjord.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct RollupCostData {
+    /// The number of zeroes in the transaction.
+    pub(crate) zeroes: u64,
+    /// The number of ones in the transaction.
+    pub(crate) ones: u64,
+    /// The size of the transaction after fastLZ compression.
+    pub(crate) fastlz_size: u32,
+}
+
+impl RollupCostData {
+    /// This takes bytes as input, creating a [`RollupCostData`] struct based on the encoded data.
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        let mut data = RollupCostData::default();
+        let mut i = 0;
+        while i < bytes.len() {
```

Comment:
the same functionality exists in gas costs, this should reuse that

Reply:
wdym / where? the counting for op data gas costs at least we want to remove, and put here instead

Reply:
https://github.com/bluealloy/revm/blob/b7b92aee232558ea48e9580bfac74d442a17babd/crates/interpreter/src/gas/calc.rs#L354-L374